File: pinyougou-web/pinyougou-user-web/src/main/webapp/js/controller/userController.js
/** 定义控制器层 */
app.controller('userController', function($scope, $timeout, baseService){

    // 定义user对象
    $scope.user = {};
    $scope.address = {};

    $scope.showName = function(){
        baseService.sendGet("/user/showName")
            .then(function(response){
                $scope.loginName = response.data.loginName;
            });
    };
    /** 用户注册 */
    $scope.save = function () {

        // 判断两次密码是否一致
        if ($scope.okPassword && $scope.user.password == $scope.okPassword){
            // 发送异步请求
            baseService.sendPost("/user/save?code=" + $scope.code, $scope.user)
                .then(function(response){
                if (response.data){
                    // 清空表单数据
                    $scope.user = {};
                    $scope.okPassword = "";
                    $scope.code = "";
                }else{
                    alert("注册失败！");
                }
            });

        }else{
            alert("两次密码不一致！");
        }
    };




    // 发送短信验证码
    $scope.sendSmsCode = function () {

        // 判断手机号码
        if ($scope.user.phone && /^1[3|4|5|7|8]\d{9}$/.test($scope.user.phone)){
            // 发送异步请求
            baseService.sendGet("/user/sendSmsCode?phone=" + $scope.user.phone)
                .then(function(response){
                if (response.data){
                    // 调用倒计时方法
                    $scope.downcount(90);

                }else{
                    alert("发送失败！");
                }
            });
        }else {
            alert("手机号码格式不正确！")
        }
    };


    $scope.smsTip = "获取短信验证码";
    $scope.disabled = false;

    // 倒计时方法
    $scope.downcount = function (seconds) {

        seconds--;

        if (seconds >= 0){
            $scope.smsTip = seconds + "秒后，重新获取！";
            $scope.disabled = true;
            // 第一个参数：回调的函数
            // 第二个参数：间隔的时间毫秒数
            $timeout(function(){
                $scope.downcount(seconds);
            }, 1000);
        }else {
            $scope.smsTip = "获取短信验证码";
            $scope.disabled = false;
        }

    };

    $scope.showAddress = function () {
        baseService.sendGet("user/showAddress?userId = " + 'itcast').then(function (response) {
            $scope.dataList = response.data;
        })
    };

    $scope.findProvinces = function () {
        $scope.address = {};
        baseService.sendGet("user/findProvinces").then(function (response) {
            $scope.province1 = response.data;
        })
    };


    $scope.$watch('address.provinceId',function (newValue, oldValue) {
        if(newValue){
            $scope.findCitiesByProvinceId(newValue);
        }else {
            $scope.cities1 = [];
        }
    });

    $scope.findCitiesByProvinceId = function(parentId){
        baseService.sendGet("/user/findCities",
            "parentId=" + parentId).then(function(response){
            $scope.cities1 = response.data;
        });
    };

    $scope.$watch('address.cityId',function (newValue) {
        if(newValue){
            $scope.findAreasBiCityId(newValue);
        }else {
            $scope.cities1 = [];
        }
    });

    $scope.findAreasBiCityId = function(parentId){
        baseService.sendGet("/user/findAreas",
            "parentId=" + parentId).then(function(response){
            $scope.areas1 = response.data;
        });
    };

    $scope.saveOrUpdate = function () {
        // 请求参数
        //alert(JSON.stringify($scope.entity));
        var url = "saveAddress"; // 添加URL
        // 判断id是否存在
        if ($scope.address.id){
            url = "update"; // 修改URL
        }

        baseService.sendPost("user/" + url,$scope.address).then(function (response) {
            if(response.data){
                location.reload()
            }else {
                alert("添加失败")
            }
        })
    };

    $scope.delete = function (id) {
        baseService.sendPost("/user/delete?id=" + id).then(function (response) {
            if(response.data){
                alert("删除成功");
                location.reload()
            }else {
                alert("删除失败")
            }
        })
    };

    $scope.updateIsDefault = function (id) {
        baseService.sendPost("/user/updateIsDefault?id=" + id ).then(function (response) {
            if(response.data){
                location.reload()
            }else {
                alert("修改失败。")
            }
        });
    }

    $scope.findAddress = function (id) {
        baseService.sendGet("/user/findAddress?id="+ id).then(function (response) {
            $scope.address = response.data;
        })
    }



    //修改密码
    $scope.updateUser = function () {
        //判断密码是否一致
        if ($scope.qrPassword && $scope.upPassword == $scope.qrPassword){
            baseService.sendPost("/user/updateUser?username=" + $scope.username + "&password=" + $scope.upPassword).then(function (response) {
                if(response.data){
                    location.href = "http://www.pinyougou.com";
                }else {
                    alert("验证码错误");
                }
            });
        }else {
            alert("密码不一致");
        }
    };

    //查询电话号码
    $scope.findPhone = function () {
        baseService.sendGet("/user/findPhone").then(function (response) {
            $scope.phoneNum = response.data;

        });
    };

    //判断验证码
    $scope.affirmCode = function (code, phone) {
        baseService.sendPost("/user/affirmCode?code=" + code).then(function (response) {
            if(response.data){
                $scope.sendCode(phone);
            }else {
                alert("验证码错误");
                window.location.reload(true);
            }
        });
    };

    //发送短信验证码
    $scope.sendCode = function (phone) {
        //发送短信验证码
        baseService.sendGet("/user/sendSmsCode?phone=" + phone).then(function (response) {
            if (response.data){
                // 调用倒计时方法
                $scope.downcount(90);

            }else{
                alert("发送失败！");
                window.location.reload(true);
            }
        });
    };

    //判断短信验证码
    $scope.messageCode = function (phone, code) {
        baseService.sendPost("/user/messageCode?phone=" + phone + "&code=" + code).then(function (response) {
            if(response.data){
                alert("验证码正确");
            }else {
                alert("验证码错误！");
                window.location.reload(true);
            }
        });
    };

    //修改电话号码
    $scope.updatePhone = function () {
        baseService.sendPost("/user/updatePhone?phone=" + $scope.newPhone).then(function (response) {
            if(!response.data){
                alert("修改失败");
                window.location.reload(true);
            }
        });
    };
});

File: pinyougou-web/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.*;
import com.pinyougou.service.*;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletRequest;
import java.io.Serializable;
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * 用户控制器
 *
 * @author lee.siu.wah
 * @version 1.0
 * <p>File Created at 2019-04-15<p>
 */
@RestController
@RequestMapping("/user")
public class UserController {

    @Reference(timeout = 10000)
    private UserService userService;
    @Reference(timeout = 10000)
    private AddressService addressService;
    @Reference(timeout = 10000)
    private ProvincesService provincesService;
    @Reference(timeout = 10000)
    private CitiesService citiesService;
    @Reference(timeout = 10000)
    private AreasService areasService;

    /**
     * 用户注册
     */
    @PostMapping("/save")
    public boolean save(@RequestBody User user, String code) {
        try {
            // 检验验证码是否正确
            boolean flag = userService.checkSmsCode(user.getPhone(), code);
            if (flag) {
                userService.save(user);
            }
            return flag;
        } catch (Exception ex) {
            ex.printStackTrace();
        }
        return false;
    }


    /**
     * 发送短信验证码
     */
    @GetMapping("/sendSmsCode")
    public boolean sendSmsCode(String phone) {
        try {
            return userService.sendSmsCode(phone);
        } catch (Exception ex) {
            ex.printStackTrace();
        }
        return false;
    }

    @GetMapping("/showAddress")
    public List<Address> showAddress(String userId) {
        try {
            return addressService.findAddressByUser(userId);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    @GetMapping("/findCity")
    public Map<String, Object> findCityByParentId(String parentId) {
        //所有省
        List<Provinces> provincesList = provincesService.findAll();
        //所有市
        List<Cities> citiesList = citiesService.findAll();
        //所有区
        List<Areas> areasList = areasService.findAll();
        Map<String, Object> map = new HashMap<>();
        map.put("provinces", provincesList);
        map.put("cities", citiesList);
        map.put("areas", areasList);
        return map;
    }

    @GetMapping("/findProvinces")
    public Map<String, Object> findProvinces() {
        //所有省
        try {
            List<Provinces> provincesList = provincesService.findAll();
            Map<String, Object> map = new HashMap<>();
            map.put("provinces", provincesList);
            return map;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    @GetMapping("/findCities")
    public Map<String, Object> findCitiesByProvinceId(String parentId) {
        Provinces provinces = provincesService.findByProvince(parentId);
        //所有市
        try {
            List<Cities> citiesList = citiesService.findCitiesByProvinceId(provinces.getProvinceId());
            Map<String, Object> map = new HashMap<>();
            map.put("cities", citiesList);
            return map;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    @GetMapping("/findAreas")
    public Map<String, Object> findAreasByCityId(String parentId) {
        Cities cities = citiesService.findByCityId(parentId);
        try {
            //所有区
            List<Areas> areasList = areasService.findAreasByCitiyId(cities.getCityId());
            Map<String, Object> map = new HashMap<>();
            map.put("areas", areasList);
            return map;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    @PostMapping("/saveAddress")
    public boolean saveAddress(@RequestBody Address address) {
        try {
            String userId = SecurityContextHolder.getContext()
                    .getAuthentication().getName();
            address.setUserId(userId);
            addressService.save(address);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    @PostMapping("/delete")
    public boolean delete(Long id){
        try {
            addressService.delete(id);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    @PostMapping("/updateIsDefault")
    public boolean updateIsDefault(Long id){
        try {
            addressService.updateIsDefault(id);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    /** 显示个人信息 */
    @GetMapping("/findInfo")
    public User findInfo(HttpServletRequest request) {
        String userId = request.getRemoteUser();
        return userService.findInfoByUserId(userId);
    }

    /** 用户信息修改 */
    @PostMapping("/userUpdate")
    public boolean userUpdate(@RequestBody User user, HttpServletRequest request){
        try {
            String username = request.getRemoteUser();
            user.setBirthday(new Date());
            user.setUsername(username);
            userService.userUpdate(user);
            return true;
        }catch (Exception e){
            e.printStackTrace();
        }
        return false;
    }

    @GetMapping("/findAddress")
    public Address findAddressById(Long id){
        return addressService.findOne(id);
    }

    @PostMapping("/update")
    public boolean update(@RequestBody Address address){
        try {
            addressService.update(address);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }



    /** 修改昵称和密码 */
    @PostMapping("/updateUser")
    public boolean updateUser(String username, String password){
        try {
            String name = SecurityContextHolder.getContext().getAuthentication().getName();
            userService.updateUser(username, password, name);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    /** 判断验证码 */
    @PostMapping("/affirmCode")
    public boolean affirmCode(String code, HttpServletRequest request){
        try {
            String oldCode = (String)request.getSession()
                    .getAttribute(VerifyController.VERIFY_CODE);
            System.out.println("oldCode = " + oldCode);
            if (code.equalsIgnoreCase(oldCode)){
                return true;
            }
            return false;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    /** 查询电话号码 */
    @GetMapping("/findPhone")
    public String findPhone(){
        try {
            String name = SecurityContextHolder.getContext().getAuthentication().getName();
            return userService.findPhone(name);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    /** 判断短信验证码 */
    @PostMapping("/messageCode")
    public boolean messageCode(String phone, String code){
        try {
            boolean flag = userService.checkSmsCode(phone, code);
            return flag;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    /** 修改电话号码 */
    @PostMapping("/updatePhone")
    public boolean updatePhone(String phone){
        try {
            String name = SecurityContextHolder.getContext().getAuthentication().getName();
            userService.updatePhone(name, phone);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}
